### DISCUSSION TITLE
Use biome as built-in linter and formatter.

### DISCUSSION THREAD

**Discussion by ianzone** [denoland/deno#25973] [Ideas]
Biome is a formatter and linter, the user can maintain one biome rule rather than eslint and prettier rules.

Biome is written in rust and should be easier to integrate with.

**Comment by bartlomieju** [comment]
Deno has its own linter (https://github.com/denoland/deno_lint) and uses multiple `dprint` (https://dprint.dev/) for formatting JS,TS,HTML,CSS,YAML and more. We're not considering using other infrastructure at the moment.

**Reply by AlDrag** [reply]
@bartlomieju would these docs be out of date then? https://docs.deno.com/runtime/fundamentals/linting_and_formatting/#formatting
As it states that Deno uses Prettier.

**Reply by bartlomieju** [reply]
Yes, it seems like a mistake in docs. I can fix it later today.

**Reply by marvinhagemeister** [reply]
Made a PR to fix this https://github.com/denoland/docs/pull/997

**Comment by acalvino4** [comment]
I think it would be great to make the various built-in utilies (test, format, lint, etc) configurable to use different tools under the hood. Totally get that deno aims to be 0-config as much as possible, so it should absolutely still default to deno's version of these things. But it would be awesome if we could make `deno test` run vitest, or `deno lint` run biome if desired.

Not sure there's a particular api that those tools would need to conform to in order for this to work, but if so, it might be worth specifying what that is to allow greater ecosystem interoperability.

**Reply by acalvino4** [reply]
actually, just checked out dprint, and see that they have a biome plugin (amongst others). I'm curious how much configuration deno allows of the underlying dprint runner, and if can be customized to use dprint's plugins.
